File: potencial_efectivo/potencial.f90
program potencial
implicit none
integer :: N, i
real(8) :: xi, xf, x, dx, v, l, pi

pi = 2.d0*asin(1.d0);
N = 20001;
l = 10;

xi = -2000.d0;
xf = 2000.d0;

dx = (xf-xi)/real(N-1, 8);

open(11, file='potencial.dat');
do i = 1, N

  x = xi + real(i-1, 8)*dx;

  if(abs(x)<= 5.d0) then
    v = sqrt(0.5d0*pi)/l*exp(x*x)*(1.d0 - erf(abs(x)));
  else
    v = sqrt(0.5d0)/l*(1.d0/abs(x) - 0.5d0/abs(x)**3);
  end if


  write(11,*) x, v

end do

close(11)

end program

File: potencial_efectivo/potencial.m
clear all
close all

l = 1;
delta = 0.5;
%eta = 0.08;

x = linspace(-20, 20, 201);

I = (abs(x)<= 5);

v_ef = zeros(1, length(x));

v_ef(I) = sqrt(0.5*pi)/l*exp(x(I).*x(I)).*(1 - erf(abs(x(I))));

v_ef(~I) = sqrt(0.5)/l*(1./abs(x(~I)) - 0.5./abs(x(~I)).^3);

v = 1./sqrt(x.^2 + delta^2);

idx = find(x==0);

eta = v_ef(idx)/v(idx);

v = eta*v;

% plot(x, v_ef, 'o-', x, v, 'o-')

salida = [x', v', v_ef'];

save('-ascii', 'l=1.dat', 'salida')

